src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,62 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+
+  def toBoxes(x: Double, y: Double, xmin: Double, xmax: Double, ymin: Double, ymax: Double, nDivisions: Int): (Int, Int) = {
+    val i = ((x - xmin) / (xmax - xmin) * (if (x.toInt == xmax) nDivisions - 1 else nDivisions)).toInt
+    val j = ((y - ymin) / (ymax - ymin) * (if (y.toInt == ymax) nDivisions - 1 else nDivisions)).toInt
+    (i, j)
+  }
+
+  def toCoords(i: Int, j: Int, xmin: Double, xmax: Double, ymin: Double, ymax: Double, nDivisions: Int): (Double, Double) = {
+    val x = xmin + ((i + 0.5) / nDivisions * (xmax - xmin))
+    val y = ymin + ((j + 0.5) / nDivisions * (ymax - ymin))
+    (x, y)
+  }
+}
+
+class RegionValueVirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double, nDivisions: Int) extends RegionValueAggregator {
+  private var boxes = Set.empty[(Int, Int)]
+
+  def seqOp(r: Region, offset: Long, missing: Boolean) {
+    if (!missing) {
+      val row = UnsafeRow.read(TTuple(Array(TFloat64(), TFloat64())), r, offset).asInstanceOf[UnsafeRow]
+      val x = row.getDouble(0)
+      val y = row.getDouble(1)
+      boxes += RegionValueVirtualDSAggregator.toBoxes(x, y, xmin, xmax, ymin, ymax, nDivisions)
+    }
+  }
+
+  def combOp(agg2: RegionValueAggregator) {
+    boxes = boxes ++ agg2.asInstanceOf[RegionValueVirtualDSAggregator].boxes
+  }
+
+  def result(rvb: RegionValueBuilder) {
+    rvb.startArray(boxes.size)
+    boxes.iterator.foreach { i =>
+      val (x, y) = RegionValueVirtualDSAggregator.toCoords(i._1, i._2, xmin, xmax, ymin, ymax, nDivisions)
```

Comment:
You can also write this:

```
  boxes.iterator.foreach { case (i, j) =>
    val (x, y) = RegionValueVirtualDSAggregator.toCoords(i, j, xmin, xmax, ymin, ymax, nDivisions)
    rvb.startTuple()
    ...
```

Minor stylistic thing, also feel free to ignore.